### ISSUE TITLE
K8s 1.22+ Support for Ambassador v1 chart

### ISSUE THREAD

**Issue by ramanNarasimhan77** [emissary-ingress/emissary#4031]
Hello,

We use seldon-core operator for deploying our ML models and expose the model serving endpoints via Ambassador

Seldon-core currently supports only Ambassador's v1 APIs and hence Ambassador v1 chart `datawire/ambassador` (latest  version:6.9.2 and appVersion:1.14.2) is only compatible with Seldon-core.  Seldon-core does not currently support the v2 API and hence is not compatible with `datawire/emissary-ingress` chart

See https://github.com/SeldonIO/seldon-core/issues/3872 

I am trying to install Seldon Core + Ambassador v1 chart in a K8s 1.22 cluster but facing problems due to [usage of deprecated/removed API ](https://github.com/emissary-ingress/emissary/files/7909821/deprecations.txt) in datawire/ambassador chart

What is the process to generate a K8s 1.22 compatible chart for datawire/ambassador chart? Fixing deprecations in ClusterRoles/ClusterRoleBindings/Role/Rolebindings is straightforward but I think all the CRD definitions need to be rebuilt to have stable version `apiextensions.k8s.io/v1` and this does not appear to be straightforward change

**Please provide guidance for rebuilding all CRD definitions.**

**Comment by alexgervais** [comment]
Hi @ramanNarasimhan77. Thanks for bringing this to our attention.
We do not plan on providing resource updates for the Ambassador v1 chart. We would encourage you (or seldon-core) to move to Emissary-ingress v2 given Ambassador 1.14's CRDs won’t work on Kubernetes 1.22+ either.

**Comment by MaxWinterstein** [comment]
**HELP**

This seems to be a problem for me here. I was running some old ambassador (0.8ish i guess) and after updating to k8s 1.22 it broke. That was expected, but i cannot get emissary 2.2.2 up. 

I guess it is some issue with old leftover mappings, for which no CRD exists any more.

Logs of ingress pod:

> emissary-ingress-5855b98445-xs77q emissary-ingress time="2022-03-30 10:02:24.7954" level=info msg="couldn't list mappings.v3alpha1.getambassador.io (will retry): conversion webhook for getambassador.io/v1, Kind=Mapping failed: no kind \"Mapping\" is registered for version \"getambassador.io/v1\" in scheme \"/go/cmd/apiext/main.go:50\"" func="github.com/datawire/ambassador/v2/pkg/kates.(*lw).List" file="/go/pkg/kates/client.go:530" CMD=entrypoint PID=1 THREAD=/watcher

> emissary-ingress-5855b98445-xs77q emissary-ingress E0330 10:02:24.795716       1 reflector.go:138] pkg/kates/client.go:439: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: conversion webhook for getambassador.io/v1, Kind=Mapping failed: no kind "Mapping" is registered for version "getambassador.io/v1" in scheme "/go/cmd/apiext/main.go:50"

Also worth to mention, I can not list mappings on namespaces that had some old mapping before, or across all namespace

```bash
k get mappings.getambassador.io -A
Error from server: conversion webhook for getambassador.io/v1, Kind=Mapping failed: no kind "Mapping" is registered for version "getambassador.io/v1" in scheme "/go/cmd/apiext/main.go:50"
```

```bash
k get mappings.getambassador.io --namespace neverhadamapping
No resources found in neverhadamapping namespace.
```

I am really kinda stuck here 😞

**Comment by MaxWinterstein** [comment]
So, we were able to solve this misery. 

It seems like there were some orphan mappings around. Those lead to a non-starting emissary, and a broken `kubectl get mappings -A`.

I downloaded the crd that contain the `emissary-apiext` deployment, deleted it, and removed the conversion webhook at `Kind: Mapping`. With this deployed, I was able to list all mappings, and after deleting the crd it deleted all mappings as well. 

Deploy the unmodified crd again and I was able to go on.

What a ride...

Now emissery-ingress started successfully and deployment of
